app/src/pages/playground/usePlaygroundPrompt.tsx

```diff
@@ -0,0 +1,211 @@
+    template: {
+      __typename: "PromptChatTemplate",
+      messages: instance.template.messages.map((m) => ({
+        content: m.content,
+        role: chatMessageRoleToPromptMessageRole(m.role),
+      })),
+    },
+    tools: instance.tools.map((t) => ({
+      __typename: "ToolDefinition",
+      definition: t.definition,
+    })),
+    // @TODO(apowell): Add description and invocationParameters
+  } satisfies Partial<PromptVersion>;
+};
+
+const getLatestPromptVersion = (
+  prompt: usePlaygroundPromptQuery$data["prompt"]
+) => {
+  if (!prompt) {
+    return null;
+  }
+  return prompt?.promptVersions?.edges?.[0]
+    ?.promptVersion as Mutable<PromptVersion> | null;
+};
+
+const query = graphql`
+  query usePlaygroundPromptQuery($promptId: GlobalID!) {
+    prompt: node(id: $promptId) {
+      ... on Prompt {
+        id
+        name
+        createdAt
+        description
+        promptVersions(first: 1) {
+          edges {
+            promptVersion: node {
+              id
+              description
+              modelName
+              modelProvider
+              invocationParameters
+              templateType
+              template {
+                __typename
+                ... on PromptChatTemplate {
+                  messages {
+                    ... on TextPromptMessage {
+                      content
+                      role
+                    }
+                  }
+                }
+              }
+              tools {
+                __typename
+                definition
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export const usePlaygroundPromptReference = (instanceId: number) => {
+  const instances = usePlaygroundContext((state) => state.instances);
+  const instance = instances.find((instance) => instance.id === instanceId);
+  if (!instance) {
+    throw new Error(`Instance ${instanceId} not found`);
+  }
+  const promptId = instance.prompt?.id;
+  const [queryReference, loadQuery, disposeQuery] =
+    useQueryLoader<usePlaygroundPromptQuery>(query);
+
+  useEffect(() => {
+    if (promptId) {
+      loadQuery({ promptId: promptId });
+      return () => disposeQuery();
+    }
+  }, [disposeQuery, loadQuery, promptId]);
+
+  return queryReference;
+};
+
+export type PromptQueryReference = NonNullable<
+  ReturnType<typeof useQueryLoader<usePlaygroundPromptQuery>>[0]
+>;
+
+export const usePlaygroundPrompt = (
+  instanceId: number,
+  queryReference: PromptQueryReference
+) => {
+  const updateInstance = usePlaygroundContext((state) => state.updateInstance);
+  const instances = usePlaygroundContext((state) => state.instances);
+  const instance = instances.find((instance) => instance.id === instanceId);
+  if (!instance) {
+    throw new Error(`Instance ${instanceId} not found`);
+  }
+  const { prompt } = usePreloadedQuery(query, queryReference);
+
+  useEffect(() => {
+    const latestPromptVersion = getLatestPromptVersion(prompt);
```

Comment:
it's this logic that needs to be repeated that I was debating whether should be pushed down into the API. Because at the end of the day there's always the current view of the prompt.

Reply:
I would prefer a latest_version key off of prompt, yeah

Reply:
Yup agree, lets add `latest_version` to the `prompt` model and then get rid of this code